src/mesido/esdl/esdl_additional_vars_mixin.py

```diff
@@ -16,8 +16,167 @@
 class ESDLAdditionalVarsMixin(CollocatedIntegratedOptimizationProblem):
     __temperature_options = {}
 
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
     def read(self):
         super().read()
+        parameters = self.parameters(0)
+        bounds = self.bounds()
+
+        # ------------------------------------------------------------------------------------------
+        # Limit available pipe classes
+        # TODO: cater for varying temperature when limiting pipe classes below
+        # Here we do a check between the available pipe classes and the demand profiles. This is to
+        # ensure that we don't have unneeded large amount of available pipe classes for pipes
+        # connected to smaller demands.
+        # TODO: add the same for electricity ones we have proper support for that in the ESDLMixin
+        if len(self.temperature_carriers().items()) == 0:
+            for asset, (
+                connected_asset,
+                _orientation,
+            ) in self.energy_system_topology.demands.items():
+                # TODO: add test case for gas once optional gas pipes are used
+                if asset in self.energy_system_components.get("gas_demand", []):
+                    try:
+                        max_demand_g_s = min(
+                            max(self.get_timeseries(f"{asset}.target_gas_demand").values),
+                            bounds[f"{asset}.Gas_demand_mass_flow"][1],
+                        )
+                    except KeyError:
+                        max_demand_g_s = bounds[f"{asset}.Gas_demand_mass_flow"][1]
+                    is_pipe_dn0_needed = True
+                    try:
+                        if min(self.get_timeseries(f"{asset}.target_gas_demand").values) > 0.0:
+                            is_pipe_dn0_needed = False
+                    except KeyError:
+                        is_pipe_dn0_needed = True
+
+                    new_pcs = []
+                    found_pc_large_enough = False
+                    for pc in self.gas_pipe_classes(connected_asset):
+                        if not found_pc_large_enough:
+                            new_pcs.append(pc)
+                            if (
+                                new_pcs[-1].maximum_discharge
+                                * self.parameters(0)[f"{connected_asset}.rho"]
```

Comment:
The density (rho) is saved in the parameters as g/m3, so the divide by 1000 is not needed.

Reply:
Nice catch!! Fixed